### DISCUSSION TITLE
Any way to disable "record_property is incompatible with xunit2" warning?

### DISCUSSION THREAD

**Discussion by petebman** [pytest-dev/pytest#10110] [Ideas] [answered]
If you use the `record_property` fixture while the junit family is set to xunit2, you get the following warning at the end of your test:

```
91: PytestWarning: record_property is incompatible with junit_family 'xunit2' (use 'legacy' or 'xunit1') 
```

I can think of two ways to interpret this warning:
  1) Any information you pass to the `record_property` fixture will be silently dropped because this fixture is incompatible with xunit2 style result XML output in some fundamental way.
  2) If you use the `record_property` fixture to add properties to the emitted junit XML, certain popular CI tools will refuse to parse the emitted XML because they use a fairly strict schema to do their XML validation.  This schema doesn't allow for arbitrary properties on a testcase element.

The 2nd interpretation is (I believe) the correct one.

My problem:  I'm not using certain popular CI tools that will refuse to parse the emitted XML.  I'm using a tool that actually handles the extra properties super well and displays them in a nice intelligent way.  I'm able to do cool things automatically like "Attach an email link to the system's engineer who is responsible for this test case" or "Automatically add a link to the product documentation that describes the thing that's failing" or even "Click this link to automatically file a bug for this failing test case." all powered by what's in the test XML.  I'd like to continue using the `xunit2` junit family but I'd also like my users to not be confused by this warning that gets printed once per test case at the end of an otherwise passing test run.

If I use `xunit-1` to suppress the warning then I get some extra properties automatically generated - `file` (this ends up being suuuuuuper long and useless because of the build tool I'm using` and `line` which is redundant to the information printed to the stack trace.  I get the same extra information that I don't want to see if I use `legacy`.

Is there some way I can tell pytest to just chill and not warn me about this?  Alternatively, can we soften the warning message a little bit to make it more clear that the problem is not with generating the XML - pytest can do that just fine - the problem is certain tools will refuse to parse the emitted XML?  I'm worried that while the warning can currently be interpreted as 2 (above) things will one day change and interpretation 1 will become correct.

Alternatively, is there a way to use `legacy` or `xunit-1` and not have the redundant properties emitted?  The same information is stored as attributes on the testcase XML element.

**Comment by The-Compiler** [comment, marked as answer]
You should be able to ignore the warning via the config (just like any other warning): https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html

**Reply by petebman** [reply]
Nice!  I'll try this - thank you.

**Reply by petebman** [reply]
`-W 'ignore:record_property is incompatible with junit_family:pytest.PytestWarning'` did the job

**Reply by markus-work** [reply]
Thank you for posting what you did, and not just saying that fixed it! 🙏
